sidh/sidh.go

```diff
@@ -12,7 +12,7 @@
 //
 // If you feel that SIDH may be appropriate for you, consult your
 // cryptographer.
-package p751sidh
```

Comment:
This package is renamed sidh, but there are many p751-specific variables and even an include of p751toolbox. If you're going to make this generic, can you remove the p751-specific code and variables? If you're going to make a generic package, it should have only the sidh-specific types and interfaces:
 type SIDHPublicKeyBob struct {}
 func (pubKey *SIDHPublicKeyBob) FromBytes(input []byte)
 func GenerateAliceKeypair(rand io.Reader) (publicKey *SIDHPublicKeyAlice, secretKey *SIDHSecretKeyAlice, err error)
 etc.

Actual implementations should be part of a curve-specific package (like sidh/p751sidh) that instantiates a function that fulfils the interfaces for SIDHPublicKeyAlice, and other structures. Unless these functions can be written generically for a specific SIDH implementation, they probably don't belong.

Reply:
It is possible to include all the p751-specific variables inside consts.go file or create another file which contains these values and sidh package simply retrieve these values from p751toolbox package. I will include it in the next commit.